libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/oauth/util/OAuthAuthorityUtils.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.auth.oauth.util;
+
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.http.HttpServletRequest;
+import com.yahoo.athenz.auth.oauth.OAuthAuthorityConsts;
+
+/**
+ * Utility class for JwtAuthority
+ */
+public class OAuthAuthorityUtils {
+
+    /**
+     * get system properties with JwtAuthority prefix
+     * @param  key property name
+     * @param  def default value
+     * @return     system property value set
+     */
+    public static final String getProperty(String key, String def) {
+        return System.getProperty(OAuthAuthorityConsts.SYSTEM_PROP_PREFIX + key, def);
+    }
+
+    /**
+     * convert CSV string to Set
+     * @param  csv       CSV string
+     * @param  delimiter CSV delimiter
+     * @return           corresponding Set object of the CSV string, or null if CSV is null or empty
+     */
+    public static final Set<String> csvToSet(String csv, String delimiter) {
+        if (csv == null || csv.isEmpty()) {
+            return null;
+        }
+        if (delimiter == null || delimiter.isEmpty()) {
+            return new HashSet<>(Arrays.asList(csv));
```

Comment:
seems to be inefficient to handle just a single string. we can just create a set and add the entry without creating additional objects.

Reply:
Thank you for pointing out. fixed: 1e7303e7